### PR TITLE
dynamic roles fix

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

**Review by ssd04 — APPROVED** [review]

File: builtInFunctions/esdtNFTTransfer.go [hunk 1]
```diff
@@ -343,23 +343,25 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
...
 	roleKey := append(roleKeyPrefix, tokenID...)
 	roles, _, err := getESDTRolesForAcnt(marshaller, account, roleKey)
 	if err != nil {
 		return false, err
 	}
 
 	dynamicRoles := [][]byte{
-		[]byte(core.ESDTRoleNFTAddURI),
-		[]byte(core.ESDTRoleNFTUpdateAttributes),
 		[]byte(core.ESDTMetaDataRecreate),
```
Comment (raduchis) [inline builtInFunctions/esdtNFTTransfer.go:354]: these roles have been newly added. No need for enableEpochsHandler for them?

File: builtInFunctions/esdtNFTTransfer.go [hunk 2]
```diff
@@ -343,23 +343,25 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
...
 		[]byte(core.ESDTRoleNFTUpdate),
 		[]byte(core.ESDTRoleModifyCreator),
 		[]byte(core.ESDTRoleModifyRoyalties),
 		[]byte(core.ESDTRoleSetNewURI),
 	}
 
+	if enableEpochsHandler.IsFlagEnabled(DynamicEsdtFlag) {
+		dynamicRoles = append(dynamicRoles, []byte(core.ESDTRoleNFTAddURI), []byte(core.ESDTRoleNFTUpdateAttributes))
```
Comment (raduchis) [inline builtInFunctions/esdtNFTTransfer.go:362]: there are no unit tests for checking the dynamicRoles
core.ESDTRoleNFTAddURI,
core.ESDTRoleNFTUpdateAttributes

File: builtInFunctions/esdtNFTTransfer.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply builtInFunctions/esdtNFTTransfer.go:354]: No, they will be added to an address only after the activation of the `DynamicNFTFlag`

File: builtInFunctions/esdtNFTTransfer.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply builtInFunctions/esdtNFTTransfer.go:362]: Added.

**Review by sasurobert — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**